src/app/CommandSender.cpp

```diff
@@ -251,10 +255,18 @@ CHIP_ERROR CommandSender::OnMessageReceived(Messaging::ExchangeContext * apExcha
 
     if (sendStatusResponse)
     {
-        StatusResponse::Send(Status::InvalidAction, apExchangeContext, false /*aExpectResponse*/);
+        bool expectingResponse                     = false;
+        Protocols::InteractionModel::Status status = Status::InvalidAction;
+        if (moreChunkedMessages && err == CHIP_NO_ERROR)
```

Comment:
What happens if `err != CHIP_NO_ERROR`?

Reply:
Then we have already called `OnErrorCallback` on line 253 and will be sending a Status response with `InvalidAction` so the other side knows we are no longer need to get more InvokeResponses

Reply:
I guess I meant that there are 4 conditions, and I can only easily understand the (true && true) case and I'm not sure if one of the 4 is not "clear" in terms of side effects. Wonder if an early return would have helped.